script.sh

```diff
@@ -78,26 +79,71 @@ else
+
+    exit 1
+  fi
+
+  jarch=$( $JAVA_HOME/bin/java -classpath . PropertyPrint os.arch )
+
+  if [ "x86_64" != "$jarch" ] ; then
+    echo "*"
+    echo "* This application bundle requires on x86-64 Java Runtime Environment (JRE/JDK)"
+    echo "*"
+    echo "* Java platform is $jarch for JAVA_HOME=$JAVA_HOME. Set JAVA_HOME to an x86-64 JRE and run this script again."
+    echo "*"
+
+    if arch -x86_64 /usr/bin/true 2> /dev/null; then
+      echo "Rosetta 2 appears to be installed, so you will already be able to run the x86-64 JRE/JDK."
+    else
+      echo "You may also need to install Rosetta 2 to run an x86-64 Java Runtime Environment."
+    fi
+
+    echo
+    echo "IPMIView.app was not built. Please read the messages above."
+
+    exit 1
+  else
+    echo "Java platform is $jarch for JAVA_HOME=$JAVA_HOME"
+  fi
+else
+  echo "Unsupported platform: $arch"
+  exit 1
 fi
-mkdir -p Contents/Resources/IPMIView/Contents/Home/bin
-tar -zxf "${LOCAL_DOWNLOAD_LOCATION}"/IPMIView*.tar* --strip=1 -C ./Contents/Resources/IPMIView/. ||
-  { echo "Something went wrong, check download of IPMIView archive" && exit 1; }
 
-echo "Linking 'java' and 'jre'..."
-ln -s "${JAVA_HOME}/bin/java" Contents/Resources/IPMIView/Contents/Home/bin/java
-rm -rf Contents/Resources/IPMIView/jre/*
-pushd Contents/Resources/IPMIView/jre/ >/dev/null &&
-  ln -s ../Contents . &&
-  popd >/dev/null || exit
+echo "Building IPMIView.app..."
+if [ -f IPMIView.app/Contents/Resources/IPMIView/IPMIView.properties ] ; then
+  echo "Saving configuration properties IPMIView.app/Contents/Resources/IPMIView/IPMIView.properties"
+  cp -a IPMIView.app/Contents/Resources/IPMIView/IPMIView.properties ./IPMIView.properties.tmp ||
+    { echo "Failed to save IPMIView.properties. Exiting."; exit 1; }
+fi
+rm -rf IPMIView.app
+mkdir IPMIView.app
+cp -a README.md Contents IPMIView.app/
+mkdir -p IPMIView.app/Contents/Resources/IPMIView/Contents/Home/bin
+tar -zxf "${LOCAL_DOWNLOAD_LOCATION}"/IPMIView*.tar* --strip=1 --exclude='*/jre/*' -C IPMIView.app/Contents/Resources/IPMIView/. ||
+  { echo "Something went wrong, check download of IPMIView archive" && exit 1; }
+ln -s "${JAVA_HOME}/bin/java" IPMIView.app/Contents/Resources/IPMIView/Contents/Home/bin/java
+if [ -f IPMIView.properties.tmp ] ; then
+  echo "Restoring IPMIView.app/Contents/Resources/IPMIView/IPMIView.properties"
+  cp -a IPMIView.properties.tmp IPMIView.app/Contents/Resources/IPMIView/IPMIView.properties ||
+    { echo "Failed to restore IPMIView.properties. Exiting."; exit 1; }
 
-echo "Copying IPMIView.app over to ~/Applications directory..."
-pushd .. >/dev/null &&
-  rsync -ar --exclude=.git --exclude=Contents/Resources/IPMIView/jre IPMIView.app ~/Applications &&
-  popd >/dev/null || exit
+  rm IPMIView.properties.tmp
+fi
 
 echo "Completed."
 echo
-echo "You can now open ~/Applications/IPMIView.app"
+echo "You can now open ./IPMIView.app or copy it into ~/Applications"
```

Comment:
can we keep this contract (that the script yields `~/Applications/IPMIView.app`) the same? any particular reason to change it?

Reply:
I'm not married to local-only build, but this is the only product I know of that _really_ wants to put itself into `~/Applications`.

There's no reason to require it to go there other than it gets into Launchpad (eventually). You could also install it into `/Applications` if you chose to.

Reply:
Understand - ultimately, it's up to the person running this script to decide where they ultimately want the app to live (in `/Applications`, or elsewhere). My point is that the change here is moving where it is *currently* stored, which might be unexpected by users who have known the script to generate the app into `~/Applications` directory. If that's something that we really really want to change, then I'd recommend we also update all other references to this, and put a clear notification to folks in the README stating that the location the app is generated to has moved.

Reply:
I'll fix this and maybe make it a completely separate PR.